File: 404/sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def loop(root, is_left):
            if root is None:
                return 0
            elif root.left is None and root.right is None and is_left:
                return root.val
            else:
                loop(root.left, True) + loop(root.right, False)

        return loop(root, False)

File: 404/SumOfLeftLeaves.scala
object SumOfLeftLeaves {
  class TreeNode(val _value: Int = 0) {
    var value: Int = _value
    var left: TreeNode = null
    var right: TreeNode = null
  }

  def sumOfLeftLeaves(root: TreeNode): Int = {
    def loop(root: TreeNode, isLeft: Boolean): Int = {
      if (root == null) 0
      else if (root.left == null && root.right == null && isLeft) root.value
      else loop(root.left, true) + loop(root.right, false)
    }
    loop(root, false)
  }

}
